pkg/core/chain/synchronizer.go

```diff
@@ -135,16 +152,12 @@ func (s *Synchronizer) ProcessBlock(m message.Message) (res []bytes.Buffer, err
 		return
 	}
 
-	s.lock.Lock()
-
-	currState := s.state
+	currState := s.getState()
 
 	if blk.Header.Height > s.highestSeenHeight {
-		s.highestSeenHeight = blk.Header.Height
+		s.setHighestSeen(blk.Header.Height)
```

Comment:
Having in mind that block is not verified at that point, could we say that any peer can send you a block with unrealistic (too-high) block height?

If so, this will make `Synchronizer.Progress` never reach 100% and as a result, all transactions sending will be permanently disabled. Could we say this is a vulnerability or I'm missing something?

@autholykos Please share your opinion too

Reply:
Correct, this however ties in to #766, and a fix for that will most likely carry over to this - for that reason, I'd argue this is out of the scope of this PR.